src/Lidarr.Api.V1/Tracks/TrackModuleWithSignalR.cs

```diff
@@ -107,15 +107,15 @@ public void Handle(TrackImportedEvent message)
         {
             foreach (var track in message.TrackInfo.Tracks)
             {
-                BroadcastResourceChange(ModelAction.Updated, track.Id);
+                BroadcastResourceChange(ModelAction.Updated, track.ToResource());
             }
         }
 
         public void Handle(TrackFileDeletedEvent message)
         {
             foreach (var track in message.TrackFile.Tracks.Value)
             {
-                BroadcastResourceChange(ModelAction.Updated, track.Id);
+                BroadcastResourceChange(ModelAction.Updated, track.ToResource());
```

Comment:
Same as artist and album above

Reply:
Is this one necessary? I suspect this change was responsible for a lot of the speedups - fetching track by id causes 4 or 5 separate dB queries per track event.  If you get chance could you try referring and see if your slowdown reappears? I never really managed to replicate locally